tests/framework/client/migrationpolicy.go

```diff
@@ -0,0 +1,85 @@
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	watch "k8s.io/apimachinery/pkg/watch"
+	rest "k8s.io/client-go/rest"
+
+	migrationsv1 "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/migrations/v1alpha1"
+
+	v1alpha1 "kubevirt.io/api/migrations/v1alpha1"
+)
+
+var createdMigrationPolicies = make([]migrationPolicyInfo, 0)
+
+type migrationPolicyInfo struct {
+	name string
+	ctx  context.Context
+}
+
+// migrationPolicies implements MigrationPolicyInterface
+type migrationPolicies struct {
+	client          rest.Interface
+	migrationPolicy migrationsv1.MigrationPolicyInterface
+}
+
+// newMigrationPolicies returns a MigrationPolicies
+func newMigrationPolicies(c *TestMigrationPolicyClient) *migrationPolicies {
+	return &migrationPolicies{
+		client:          c.client.RESTClient(),
+		migrationPolicy: c.client.MigrationPolicies(),
+	}
+}
+
+func (c *migrationPolicies) Clean() {
+	for _, mp := range createdMigrationPolicies {
+		err := c.Delete(mp.ctx, mp.name, v1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			panic(err)
+		}
+	}
+}
+
+func (c *migrationPolicies) Get(ctx context.Context, name string, options v1.GetOptions) (result *v1alpha1.MigrationPolicy, err error) {
+	return c.migrationPolicy.Get(ctx, name, options)
+}
+
+func (c *migrationPolicies) List(ctx context.Context, opts v1.ListOptions) (result *v1alpha1.MigrationPolicyList, err error) {
+	return c.migrationPolicy.List(ctx, opts)
+}
+
+func (c *migrationPolicies) Watch(ctx context.Context, opts v1.ListOptions) (watch.Interface, error) {
+	return c.migrationPolicy.Watch(ctx, opts)
+}
+
+func (c *migrationPolicies) Create(ctx context.Context, migrationPolicy *v1alpha1.MigrationPolicy, opts v1.CreateOptions) (result *v1alpha1.MigrationPolicy, err error) {
+	created, err := c.migrationPolicy.Create(ctx, migrationPolicy, opts)
+	if err == nil && opts.DryRun == nil {
+		createdMigrationPolicies = append(createdMigrationPolicies, migrationPolicyInfo{created.Name, ctx})
+	}
+	return created, err
+}
+
+func (c *migrationPolicies) Update(ctx context.Context, migrationPolicy *v1alpha1.MigrationPolicy, opts v1.UpdateOptions) (result *v1alpha1.MigrationPolicy, err error) {
+	return c.migrationPolicy.Update(ctx, migrationPolicy, opts)
+}
+
+func (c *migrationPolicies) UpdateStatus(ctx context.Context, migrationPolicy *v1alpha1.MigrationPolicy, opts v1.UpdateOptions) (result *v1alpha1.MigrationPolicy, err error) {
+	return c.migrationPolicy.UpdateStatus(ctx, migrationPolicy, opts)
+}
+
+func (c *migrationPolicies) Delete(ctx context.Context, name string, opts v1.DeleteOptions) error {
```

Comment:
Maybe in delete we would like to take the policy out of `createdMigrationPolicies`?
But on the other hand this is just a pointer and shouldn't really effect test performance / usa a lot of memory

Reply:
We could, but since  `Clean()` ignore notFound errors, I think we can leave it as is. At most we try to remove it twice

Reply:
Right, but it still has two disadvantages:
* Memory leak (although very small since we're only dealing with pointers)
* Overhead (since you make an API call for nothing)

Of course these two are far from being significant. Your call :)

Reply:
Agreed, In this case I think I will move `createdMigrationPolicies` from slice to map in order to not iterate over it for every `Delete`. WDYT?

Reply:
Sounds good to me :+1: 
I would also remove it from the map if I know a policy is already deleted